File: totosuki/ABC/C/352.py
m,*A=0,
j,*B=0,
for i in range(int(input())):
 a,b=map(int,input().split());A+=[a];B+=[b]
 if m<b-a:m=a-b;j=i
a=0
for i in range(len(A)):
 if i!=m:a+=A[i]
print(a+B[j])

File: OCH_Tao/ABC/C/352.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
input.shift();
let X = 0;
let H = 0;
for (const i of input) {
  const [A, B] = i.split(" ").map(x => parseInt(x, 10));
  H = Math.max(H, B - A);
  X += A;
}
console.log(X + H);